### PR TITLE
[MRG] Add a PyTorch solver to ot.dr.wda (#806)

### PR DESCRIPTION
**Pull request by deeb01**
## Types of changes

New feature (non-breaking change which adds functionality).

## Motivation and context / Related issue

Addresses #806, following @rflamary's request for `solver='torch'` alongside the default `solver='autograd'`.

`solver='torch'` runs PyTorch autodiff with Riemannian gradient descent on Stiefel — QR retraction, backtracking with an adaptive initial step — mirroring pymanopt's `SteepestDescent` so both solvers target the same optimum. It needs only torch, and accepts torch tensors directly, keeping device and dtype.

`ot.dr` dependencies are now imported optionally, each function raising an `ImportError` naming what it needs. **This changes `import ot.dr` from raising to succeeding on a partial install** — flagging it in case you prefer the old behaviour.

On speed: pymanopt is 0.6–1.8% of runtime and this solver is 1.0–1.75× faster, so the dependency choice is the real benefit. A frozen-plan gradient for a larger win stalls at a worse objective; numbers on the issue.

Also raises a clear `ValueError` when the between-class cost underflows to zero, previously a divide-by-zero.

## How has this been tested (if it applies)

New tests assert the torch objective and gradient match the autograd ones (rtol 1e-10, 1e-8) and that both solvers reach a comparable objective from the same start; ratio 0.997–1.004 across n, k, C and `sinkhorn_method`. Also covered: tensor input dtype, `sinkhorn_log`, input not mutated, error path.

`test_dr.py` 16 passed; full suite 2668 passed, 62 skipped, 6 xfailed; pre-commit clean.

## PR checklist

- [x] I have read the [**CONTRIBUTING**](https://pythonot.github.io/contributing.html) document.
- [x] The documentation is up-to-date with the changes I made (check build artifacts).
- [x] All tests passed, and additional code has been **covered with new tests**.
- [x] I have added the PR and Issue fix to the [**RELEASES.md**](RELEASES.md) file.

### CODE REVIEW

File: ot/dr.py [hunk 1]
```diff
@@ -79,6 +111,184 @@ def split_classes(X, y):
...
+    ui = torch.zeros(M.shape[0], dtype=M.dtype, device=M.device)
+    vi = torch.zeros(M.shape[1], dtype=M.dtype, device=M.device)
+    log_w1, log_w2 = torch.log(w1), torch.log(w2)
+    for _ in range(k):
+        vi = log_w2 - torch.logsumexp(Mr + ui[:, None], 0)
+        ui = log_w1 - torch.logsumexp(Mr + vi[None, :], 1)
+    return torch.exp(ui[:, None] + Mr + vi[None, :])
+
+
+def _stiefel_retract(P, X):
```
Comment (rflamary) [inline ot/dr.py]:
```suggestion
def _stiefel_retract_torch(P, X):
```
